uswds/uswds-compile
# USWDS-Compile - POAM: May '24

# Summary

Updated non-vulnerable dependencies including the USWDS package.

## Snyk errors

Updated snyk ignore to resolve the following issues:

```
npx snyk ignore --id="SNYK-JS-BRACES-6838727" --reason="No available upgrade or patch" 
npx snyk ignore --id="SNYK-JS-INFLIGHT-6095116" --reason="No available upgrade or patch"
npx snyk ignore --id="SNYK-JS-MICROMATCH-6838728" --reason="No available upgrade or patch"
npx snyk ignore --id="SNYK-JS-POSTCSS-5926692" --reason="No available upgrade or patch" 
npx snyk ignore --id="SNYK-JS-ANSIREGEX-1583908" --reason="Upgrading Gulp causes regression in asset compilation"
```

## Testing instructions

1. Install this branch on site
    - On Site's `main` branch, run: 
  ```bash
npm install "https://github.com/uswds/uswds-compile/tree/cm-POAM-may-2024" --save
```
2. Run gulp sass scripts and ensure compile runs as expected without error.
3. Ensure no when package is used to compile site
    1. Visit [demo repo](https://github.com/uswds/uswds-site/pull/2681)
    2. Run gulp compile commands and confirm they run without error
    3. Visit [site preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-demo-compile-101/) and confirm there are no regressions

## Dependency updates

| Dependency | Old version | New version |
|--------|--------|--------|
| `sass-embedded` | 1.74.1 | 1.77.0 |
